mgo.go

```diff
@@ -0,0 +1,106 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+const (
+	dbName             = "db"
+	monthlyInfoColName = "mi"
+	maxValue           = math.MaxFloat64
+)
+
+//collection is a private interface to create a mongo's ReplaceOne method and their signatures to be used and tested.
+type collection interface {
+	ReplaceOne(ctx context.Context, filter interface{}, replacement interface{}, opts ...*options.ReplaceOptions) (*mongo.UpdateResult, error)
+}
+
+//Client is a mongodb Client instance
+type Client struct {
+	mgoClient *mongo.Client
+	col       collection
+}
+
+//NewClient instantiates a new client, but will not connect to the specified URL. Please use Client.Connect before using the client.
+func NewClient(url string) (*Client, error) {
+	client, err := mongo.NewClient(options.Client().ApplyURI(url))
+	if err != nil {
+		return nil, err
+	}
+	return &Client{mgoClient: client}, nil
+}
+
+//Connect establishes a connection to MongoDB using the previously specified URL
+func (c *Client) Connect() error {
+	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+	defer cancel()
+	if err := c.mgoClient.Connect(ctx); err != nil {
+		return fmt.Errorf("error connection with mongo:%q", err)
+	}
+	c.col = c.mgoClient.Database(dbName).Collection(monthlyInfoColName)
+	return nil
+}
+
+//Disconnect closes the connections to MongoDB. It does nothing if the connection had already been closed.
+func (c *Client) Disconnect() error {
+	if c.col == nil {
+		return nil
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+	defer cancel()
+	c.col = nil
+	return c.mgoClient.Disconnect(ctx)
+}
+
+// Store processes and stores the crawling results.
+func (c *Client) Store(cr CrawlingResult) error {
+	if c.col == nil {
+		return fmt.Errorf("Collection is nil")
```

Comment:
Por favor substituir por uma mensagem de erro mais descritiva: "Client is not connected."